05-UI-ReactJS/Topic5.html

```diff
@@ -0,0 +1,224 @@
+<!DOCTYPE html>
+<html class="no-js" lang="">
+
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="x-ua-compatible" content="ie=edge">
+    <title>Topic 5</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+  	<script src="https://unpkg.com/react@latest/dist/react.js"></script>
+    <script src="https://unpkg.com/react-dom@latest/dist/react-dom.js"></script>
+    <script src="https://unpkg.com/babel-standalone@6.15.0/babel.min.js"></script>
+</head>
+<body>
+	<div id="root">
+		
+	</div>
+<script>
+	class movie{
+
+		constructor(title, year, duration){
+			this.title = title;
+			this.year = year;
+			this.duration = duration;
+		}
+
+	}
+</script>
+
+<script type="text/babel">
+	class Movie extends React.Component{
+		render(){
+		 return <li>
+		 			Title: {this.props.title} 
+		 			| Year: {this.props.year}
+		 			| Duration {this.props.duration}
+		 			<br />
+		 		</li>
+		}		
+	}
+
+	class MoviesAdministration extends React.Component{
+		constructor(props){
+			super(props);
+			this.state = {
+				title: '',
+				year: '', 
+				duration: '',
+				movieList: [],
+				newTitle: '',
+				newYear: '',
+				newDuration: '',
+				isEdited: false,
+			};
+			this.handleChangeTitle = this.handleChangeTitle.bind(this),
+			this.handleChangeYear = this.handleChangeYear.bind(this),
+			this.handleChangeDuration = this.handleChangeDuration.bind(this),
+
+			this.handleTitleEdit = this.handleTitleEdit.bind(this),
+			this.handleYearEdit = this.handleYearEdit.bind(this),
+			this.handleDurationEdit = this.handleDurationEdit.bind(this),
+
+			this.addMovie = this.addMovie.bind(this),
+			this.changeFormStyle = this.changeFormStyle.bind(this),
+			this.editMovie = this.editMovie.bind(this),
+			this.deleteMovie = this.deleteMovie.bind(this);
+		}
+		//Here starts handlers to set the state of a movie *
+		handleChangeTitle(event) {	  
+		    this.setState({
+		      title: event.target.value
+		    })
+		}
+		handleChangeYear(event){
+			this.setState({
+				year: event.target.value
+			})
+		}
+		handleChangeDuration(event){
+			this.setState({
+				duration: event.target.value
+			})
+		}
+		addMovie(event) {
+			event.preventDefault();
+			let movie = new Movie();
+			movie.title = this.state.title;
+			movie.year = this.state.year;
+			movie.duration = this.state.duration;
+		    this.state.movieList.push(movie);
```

Comment:
es mejor no cambiar directamente this.state, usar una variable auxiliar y después llamar a this.setState()